### PR TITLE
refactor: add network to persisted profile

### PR DESCRIPTION
**Pull request by nicole-obrien**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/network/constants/network.constant.ts [hunk 1]
```diff
@@ -1,72 +1,60 @@
...
-            name: 'IOTA',
-            protocol: NetworkProtocol.IOTA,
-            type: NetworkType.Mainnet,
+export const NETWORK: Readonly<{ [key in NetworkId]?: INetwork }> = {
+    [NetworkId.Iota]: {
+        id: NetworkId.Iota,
+        name: 'IOTA',
+        protocol: {
+            version: 1,
+            networkName: 'iota',
```
Comment (Tuditi) [inline packages/shared/lib/core/network/constants/network.constant.ts:11]: Maybe worth using the `NetworkProtocol` enum here and on l.29?

File: packages/shared/lib/core/network/constants/network.constant.ts [hunk 2]
```diff
@@ -1,72 +1,60 @@
...
+            belowMaxDepth: 15,
             rentStructure: {
-                vByteCost: 500,
-                vByteFactorData: 10,
-                vByteFactorKey: 1,
+                vByteCost: 100,
+                vByteFactorData: 1,
+                vByteFactorKey: 10,
             },
+            tokenSupply: '1813620509061365',
```
Comment (Tuditi) [inline packages/shared/lib/core/network/constants/network.constant.ts:38]: Let's move this magic number to a constant.

File: packages/shared/lib/core/network/constants/network.constant.ts [hunk 3]
```diff
@@ -1,72 +1,60 @@
...
+            belowMaxDepth: 15,
             rentStructure: {
-                vByteCost: 500,
-                vByteFactorData: 10,
-                vByteFactorKey: 1,
+                vByteCost: 100,
+                vByteFactorData: 1,
+                vByteFactorKey: 10,
             },
+            tokenSupply: '1450896407249092',
```
Comment (Tuditi) [inline packages/shared/lib/core/network/constants/network.constant.ts:56]: Same here

**Review by Tuditi — CHANGES_REQUESTED** [review]
Nice work! I left some questions and requested minor changes. :relaxed:

File: packages/shared/lib/core/network/constants/network.constant.ts (hunk 1 above)
Comment (nicole-obrien) [reply packages/shared/lib/core/network/constants/network.constant.ts:11]: Plan is to reduce use of that so I can eventually remove it

File: packages/shared/lib/core/profile/actions/active-profile/saveActiveProfile.ts [hunk 4]
```diff
@@ -10,6 +10,7 @@ export function saveActiveProfile(): void {
             id: _activeProfile?.id,
             name: _activeProfile?.name,
             type: _activeProfile?.type,
+            network: _activeProfile?.network,
             networkProtocol: _activeProfile?.networkProtocol,
             networkType: _activeProfile?.networkType,
```
[the remark this answers is not in the document]
Comment (nicole-obrien) [reply packages/shared/lib/core/profile/actions/active-profile/saveActiveProfile.ts:15]: They are going to be removed

**Pull request merged**